### PR TITLE
[FLINK-21273][coordination] Remove unused ExecutionVertexSchedulingRe…

### PR DESCRIPTION
**Pull request by KarmaGYZ**
…quirements

## What is the purpose of the change

Remove unused ExecutionVertexSchedulingRequirements.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

cc @zhuzhurk

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/ExecutionSlotAllocator.java [hunk 1]
```diff
@@ -29,10 +29,10 @@
     /**
      * Allocate slots for the given executions.
      *
-     * @param executionVertexSchedulingRequirements The requirements for scheduling the executions.
+     * @param executionVertexIds The requirements for scheduling the executions.
```
Comment (zhuzhurk) [inline flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/ExecutionSlotAllocator.java:32]: "The requirements for scheduling the executions" -> "Execution vertices to allocate slots for"

File: flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SlotSharingExecutionSlotAllocator.java [hunk 2]
```diff
@@ -115,15 +115,11 @@
      *       returns the results.
      * </ol>
      *
-     * @param executionVertexSchedulingRequirements the requirements for scheduling the executions.
+     * @param executionVertexIds the requirements for scheduling the executions.
```
Comment (zhuzhurk) [inline flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SlotSharingExecutionSlotAllocator.java:118]: the comment is outdated

**Review by zhuzhurk — COMMENTED** [review]
Thanks for creating this PR! @KarmaGYZ 
It looks good to me except for 2 minor comments.

**Comment by KarmaGYZ** [comment]
Thanks for the review @zhuzhurk . I've addressed your comments.

**Review by zhuzhurk — APPROVED** [review]
Thanks for addressing the comments @KarmaGYZ 
The change looks good to me. 
CI already gives green and I will merge it.

**Pull request merged**